tests/oracle.rs

```diff
@@ -0,0 +1,372 @@
+
+    let mut oracle_account =
+        get_oracle_account(oracle_pub_key, oracle_account_id, oracle_storage_slots);
+
+    let [oracle_data_1, oracle_data_2, oracle_data_3, oracle_data_4] = mock_oracle_data();
+
+    let oracle_data_word_1 = oracle_data_1.to_word();
+    let oracle_data_word_2 = oracle_data_2.to_word();
+    let oracle_data_word_3 = oracle_data_3.to_word();
+    let oracle_data_word_4 = oracle_data_4.to_word();
+
+    // CONSTRUCT AND EXECUTE TX
+    // --------------------------------------------------------------------------------------------
+    let tx_context = TransactionContextBuilder::new(oracle_account.clone()).build();
+    let executor =
+        TransactionExecutor::new(Arc::new(tx_context.clone()), Some(oracle_auth.clone()));
+    let block_ref = tx_context.tx_inputs().block_header().block_num();
+
+    // Create transaction script to write the data to the oracle account
+    let tx_script_code = format!(
+        "{}",
+        WRITE_DATA_TX_SCRIPT
+            .replace("{1}", &word_to_masm(oracle_data_word_1))
+            .replace("{2}", &word_to_masm(oracle_data_word_2))
+            .replace("{3}", &word_to_masm(oracle_data_word_3))
+            .replace("{4}", &word_to_masm(oracle_data_word_4))
+            .replace(
+                "[1]",
+                &format!("{}", oracle_account.code().procedures()[1].mast_root()).to_string()
+            )
+    );
+
+    let assembler = TransactionKernel::assembler();
+    let tx_script = TransactionScript::compile(
+        tx_script_code,
+        [],
+        // Add the oracle account's component as a library to link
+        // against so we can reference the account in the transaction script.
+        assembler
+            .with_library(ORACLE_COMPONENT_LIBRARY.as_ref())
+            .expect("adding oracle library should not fail")
+            .clone(),
+    )
+    .unwrap();
+    let txn_args = TransactionArgs::with_tx_script(tx_script);
+
+    let executed_transaction = executor
+        .execute_transaction(oracle_account.id(), block_ref, &[], txn_args)
+        .unwrap();
+
+    oracle_account
+        .apply_delta(executed_transaction.account_delta())
+        .unwrap();
+
+    // check that the oracle account has successfully been updated with the correct values
+    assert_eq!(
+        oracle_account.storage().slots()[1].value(),
+        oracle_data_word_1
+    );
+    assert_eq!(
+        oracle_account.storage().slots()[2].value(),
+        oracle_data_word_2
+    );
+    assert_eq!(
+        oracle_account.storage().slots()[3].value(),
+        oracle_data_word_3
+    );
+    assert_eq!(
+        oracle_account.storage().slots()[4].value(),
+        oracle_data_word_4
+    );
+}
+
+#[test]
+fn test_oracle_read() {
+    //  SETUP
+    // --------------------------------------------------------------------------------------------
+    let (oracle_pub_key, _) = get_new_pk_and_authenticator();
```

Comment:
Can you add a few more comments here? Then it might be easier to understand for their developer why we do what we do?